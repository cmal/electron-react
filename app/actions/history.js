// @flow
// import type { GetState, Dispatch } from '../reducers/types';

// export const ON_FOCUS = 'ON_FOCUS';
// export const ON_BLUR = 'ON_BLUR';
export const NO_ACTION = 'NO_ACTION';

export function noAction() {
  return {
    type: NO_ACTION
  };
}

// export function onBlur() {
//   return {
//     type: ON_BLUR
//   };
// }
