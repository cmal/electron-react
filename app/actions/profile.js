// @flow
// import type { GetState, Dispatch } from '../reducers/types';

export const NO_ACTION = 'NO_ACTION';

export function onAction() {
  return {
    type: NO_ACTION
  };
}
