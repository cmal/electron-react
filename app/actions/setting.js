// @flow
// import type { GetState, Dispatch } from '../reducers/types';

export const NO_ACTION = 'NO_ACTION';

export function noAction() {
  return {
    type: NO_ACTION
  };
}
