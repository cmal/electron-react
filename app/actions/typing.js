// @flow
// import type { GetState, Dispatch } from '../reducers/types';

export const LOAD_CLIPBOARD = 'LOAD_CLIPBOARD';

export function loadClipboard() {
  return {
    type: LOAD_CLIPBOARD
  };
}
