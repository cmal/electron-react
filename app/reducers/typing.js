// @flow
import { clipboard } from 'electron';
import { LOAD_CLIPBOARD } from '../actions/typing';
import type { Action, TypingState } from './types';
import { typingState } from './types';

export default function typing(
  state: TypingState = typingState,
  action: Action
) {
  switch (action.type) {
    case LOAD_CLIPBOARD:
      return {
        ...state,
        text: clipboard.readText()
      };
    default:
      return state;
  }
}
