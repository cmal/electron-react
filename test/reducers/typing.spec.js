import typing from '../../app/reducers/typing';
import { LOAD_CLIPBOARD } from '../../app/actions/typing';

describe('reducers', () => {
  describe('typing', () => {
    it('should handle initial state', () => {
      expect(typing(undefined, {})).toMatchSnapshot();
    });

    it('should handle LOAD_CLIPBOARD', () => {
      expect(typing(1, { type: LOAD_CLIPBOARD })).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(typing(1, { type: 'unknown' })).toMatchSnapshot();
    });
  });
});
