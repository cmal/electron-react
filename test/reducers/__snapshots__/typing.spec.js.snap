// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers typing should handle LOAD_CLIPBOARD 1`] = `1`;

exports[`reducers typing should handle initial state 1`] = `
Object {
  "counter": 0,
  "typing": Object {
    "text": "demo",
    "typingText": "",
  },
}
`;

exports[`reducers typing should handle unknown action type 1`] = `1`;
